### PR TITLE
Add Title Case Conversion

### PR DESCRIPTION
**Pull request by 0APOCALYPSE0**
### Describe your change:

convert lower or upper case string to title case

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: strings/title.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+def convert_word(word: str) -> str:
```
Comment (cclauss) [inline strings/title.py:1]: Let's have a self-documenting function name.
```suggestion
def to_title_case(word: str) -> str:
```

Comment (0APOCALYPSE0) [reply strings/title.py:1]: @cclauss Hey, Actually I am using two methods:
convert_word will convert a single word to a title case word ex: "dog" to "Dog"
THEN main method to_title_case will take all the converted words and will join them together to create a sentence.

so just wanted to let you know we already have a method name to_title_case so maybe I can give some other name to convert_word method like to_title_word or something else

Comment (cclauss) [reply strings/title.py:1]: Let's keep the word one short: `to_title_case()` and the sentence one longer: `sentence_to_title_case()` and we can leave the word `convert` out of the function names.

File: strings/title.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    'Aakash Giri'
+
+    >>> to_title_case("aAkAsH gIrI")
+    'Aakash Giri'
+    """
+
+    words = input_str.split()
+    title_case_str = [convert_word(word) for word in words]
+
+    return " ".join(title_case_str)
```
Comment (cclauss) [inline strings/title.py:54]:
```suggestion
    return " ".join(convert_word(word) for word in input_str.split())
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**